mlrun/serving/states.py

```diff
@@ -1218,6 +1243,89 @@ def __init__(
         self.raise_exception = raise_exception
         self.shape = "folder"
 
+    def add_shared_model_proxy(
```

Comment:
I still think in terms of naming, it probably makes sense to talk about a shared model, or a model proxy, but not a shared model proxy. Wdyt?

Reply:
Maybe I've been reading it wrong, as shared (model proxy) rather than (shared model) proxy.

Reply:
there's the add_shared_model function which adds the models, so users are not likely to confuse since their code will first add the shared models, then add the proxies as part of the graph construction.